src/app/api/dashboard/resumes/route.js

```diff
@@ -0,0 +1,170 @@
+import { NextResponse } from "next/server";
+import { db } from "@/utils/firebase";
+import {
+  collection,
+  getDocs,
+  doc,
+  updateDoc,
+  deleteDoc,
+  getDoc,
+  orderBy,
+  endBefore,
+  limitToLast,
+  limit,
+  getCountFromServer,
+  query,
+  where,
+} from "firebase/firestore";
+import { authenticate } from "@/utils/auth";
+import { AUTH } from "@/data/admin/dashboard";
+
+export const GET = async (req) => {
+  const direction = req.nextUrl.searchParams.get("direction");
+  const index = req.nextUrl.searchParams.get("index");
+  const size = req.nextUrl.searchParams.get("size");
+  const first = req.nextUrl.searchParams.get("first");
+  const last = req.nextUrl.searchParams.get("last");
+
+  const res = NextResponse;
+  const { auth, message } = await authenticate(AUTH.GET);
+
+  if (auth !== 200) {
+    return res.json(
+      { message: `Authentication Error: ${message}` },
+      { status: auth },
+    );
+  }
+  const output = [];
+  try {
+    let snapshot;
+
+    if (direction === "next" && last !== "undefined") {
+      const lastDocument = await getDoc(doc(db, "resumes", last));
+
+      snapshot = await getDocs(
```

Comment:
Check with Andrew on infinite scrolling, he implemented it for the other stuff so make sure this is also infinite scrolling